File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    private void swap(int[] v, int i, int j) {
        v[j] = v[i];
        return;
    }
    public int removeDuplicates(int[] nums) {
        int j = 1, count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i-1]) {
                count++;
            } else {
                count = 1;
            }

            if (count > 2) {
                // save j here
            } else {
                swap(nums, i, j);
                j++;
            }
        }
        return j; // TC: O(n), SC: O(1)
    }
}

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            int cntI = count(nums.begin(), nums.end(), nums[i]);
            if (cntI > 2) {
                int j = i + 1;
                while (j < nums.size() && nums[j] == nums[i] && cntI > 2) {
                    nums.erase(nums.begin() + j);
                    cntI--;
                }
            }
        }
        return nums.size();
    }
};